tag-release.sh

```diff
@@ -35,57 +12,70 @@ function current_branch() {
   git rev-parse --abbrev-ref @
 }
 
-if ! branch_exists "${release_branch}" ; then
-  git checkout master
-  if [ ! "$(current_branch)" = master ] ; then
-    echo "Must be on master branch"
-    exit 7
-  fi
-  # Ensure local master is up-to-date by pulling its latest version from origin
-  # and fast-forwarding local master:
-  git fetch origin master
-  git merge --ff-only origin/master
-  # Create the release branch:
-  git push origin master:"${release_branch}"
+#if [[ ! "$(git remote get-url origin)" =~ ^git@github.com:weaveworks/eksctl(\.git)?$ ]] ; then
+#  echo "Invalid origin: $(git remote get-url origin)"
+#  exit 3
+#fi
+
+v=$(go run pkg/version/generate/release_generate.go print-version)
+
+release_branch="release-${v%.*}"  # e.g.: 0.2.0 -> release-0.2
+if ! [[ "${release_branch}" =~ ^release-[0-9]+\.[0-9]+$ ]] ; then
+  echo "Invalid release branch: ${release_branch}"
+  exit 3
 fi
 
-# Ensure local release branch is up-to-date by pulling its latest version from
-# origin and fast-forwarding the local branch:
-git fetch origin "${release_branch}"
-git checkout "${release_branch}"
 if [ ! "$(current_branch)" = "${release_branch}" ] ; then
   echo "Must be on ${release_branch} branch"
   exit 5
 fi
-git merge --ff-only origin/"${release_branch}"
 
-RELEASE_NOTES_FILE="docs/release_notes/${v}.md"
+# Ensure local release branch is up-to-date by pulling its latest version from
+# origin and fast-forwarding the local branch:
+git pull --ff-only origin "${release_branch}" || echo "${release_branch} not found in origin. Will push new branch upstream"
 
+RELEASE_NOTES_FILE="docs/release_notes/${v}.md"
 if [[ ! -f "${RELEASE_NOTES_FILE}" ]]; then
   echo "Must have release notes ${RELEASE_NOTES_FILE}"
   exit 6
 fi
 
 export RELEASE_GIT_TAG="${v}"
 
-go generate ./pkg/version
-
+# Update eksctl version by removing the pre-release id
+go run pkg/version/generate/release_generate.go release
 git add ./pkg/version/release.go
 git add "${RELEASE_NOTES_FILE}"
 
-m="Tag ${v} release"
+m="Release ${v}"
 
 git commit --message "${m}"
-
-git fetch --force --tags origin
-
 git push origin "${release_branch}"
 
-# Update the site by putting everything from master into the docs branch
-git push --force origin "${release_branch}":docs
-
 # Create the release tag and push it to start release process
 git tag --annotate --message "${m}" --force "latest_release"
 git tag --annotate --message "${m}" "${v}"
+git push origin "${v}"
+
+# Update the site by putting everything from master into the docs branch
+git push --force origin "${release_branch}":docs
+
+# Check if need to bump version in master
+git checkout master
+if [ ! "$(current_branch)" = master ] ; then
```

Comment:
Do we need this check? When can `git rev-parse --abbrev-ref @` return something other than the checked out branch?

Reply:
I think if for some reason you have uncommited changes the checkout can fail. I kept this from the existing script https://github.com/weaveworks/eksctl/blob/master/tag-release.sh#L39

Do you think it's better to remove it? I'm thinking it could be a good fail safe.

Reply:
>I kept this from the existing script https://github.com/weaveworks/eksctl/blob/master/tag-release.sh#L39
I see. Let's keep it then.